core/src/main/java/net/adoptopenjdk/icedteaweb/resources/ResourceHandler.java

```diff
@@ -34,8 +33,8 @@
 class ResourceHandler {
 
     private static final Logger LOG = LoggerFactory.getLogger(ResourceHandler.class);
-    private static final Executor localExecutor = new ThreadPoolExecutor(0, Runtime.getRuntime().availableProcessors() * 2,
-            10L, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+
+    private static final Executor localExecutor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 4, new DaemonThreadFactory());
```

Comment:
Using a fixed thread pool for the download is not optimal.
Because after the initial download the threads are no longer used. Never the less the fixed thread pool will keep them idle until the end of the application

Reply:
I'm with you. This executor was just wrong and only executed one task in parallel. This is a quick fix to check that multiple downloads work in parallel. And yes, it works ;)